From 9505285498d443a51c86d042e09acc2036a92479 Mon Sep 17 00:00:00 2001
From: Tomasz Figa <t.figa@samsung.com>
Date: Fri, 1 Mar 2013 18:28:31 +0100
Subject: [PATCH 0109/1302] clk: samsung: pll: Do not return error codes in
 round_rate callback

This patch modifies pll35xx round_rate callback to return current
frequency instead of error codes to avoid problems caused by clock core
using error codes as frequencies.

Signed-off-by: Tomasz Figa <t.figa@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/clk/samsung/clk-pll.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/samsung/clk-pll.c b/drivers/clk/samsung/clk-pll.c
index 26292d4..1abdfba 100644
--- a/drivers/clk/samsung/clk-pll.c
+++ b/drivers/clk/samsung/clk-pll.c
@@ -81,14 +81,14 @@ static long samsung_pll35xx_round_rate(struct clk_hw *hw,
 
 	if (!pms) {
 		pr_err("%s: no pms table passed", __func__);
-		return -ENOTSUPP;
+		return samsung_pll35xx_recalc_rate(hw, *prate);
 	}
 
 	for (i = 0; pms[i].f_out != F_OUT_INVAL; i++)
 		if (drate >= pms[i].f_out)
 			return pms[i].f_out;
 
-	return -EINVAL;
+	return samsung_pll35xx_recalc_rate(hw, *prate);
 }
 
 static int samsung_pll35xx_set_rate(struct clk_hw *hw, unsigned long drate,
-- 
1.8.3.2

